### PR TITLE
Fix CAS with LDAP authentication

### PR DESCRIPTION
**Pull request by Utopism**
Please review that Pull Request that makes CAS authentication with LDAP work.
Fixes https://mantis.ilias.de/view.php?id=24931

modified :         Services/CAS/classes/class.ilAuthProviderCAS.php

Thanks !

### CODE REVIEW

**Comment by klees** [comment]
Hi @Utopism, welcome at ILIAS!

**Comment by Amstutz** [comment]
Hi @smeyer-ilias 

Any update on this?

Kindly the TB

**Comment by mjansenDatabay** [comment]
@smeyer-ilias Could you please give any feedback here? It is a valid option to close this PR if you have any objections. Otherwise feedback would be highly appreciated.

File: Services/CAS/classes/class.ilAuthProviderCAS.php [hunk 1]
```diff
@@ -121,17 +125,32 @@ public function doAuthentication(\ilAuthStatus $status)
 	protected function handleLDAPDataSource(\ilAuthStatus $status)
 	{
 		include_once './Services/LDAP/classes/class.ilLDAPServer.php';
+		// Fixed : AUTH_CAS -> AUTH_LDAP
 		$server = ilLDAPServer::getInstanceByServerId(
-			ilLDAPServer::getDataSource(AUTH_CAS)
+			ilLDAPServer::getDataSource(AUTH_LDAP)
```
Comment (smeyer-ilias) [inline Services/CAS/classes/class.ilAuthProviderCAS.php:130]: This modification shouldn't be necessary. 
Maybe this works in your case but the ldap server should be configured as datasource for the authentication mode "CAS".

**Review by smeyer-ilias — COMMENTED** [review]
Thank you very much for your contribution. 
If have to check the combination of CAS in combination with LDAP but unfortunately I have to create a new CAS-Authentication-Test instance before.
I cannot accept the PR because it relies on changes for the authentication mode of the user account.

**Comment by Amstutz** [comment]
Ill close this one since declined by the maintainer.

**Pull request closed without merging**